src/third_party/CMakeLists.txt

```diff
@@ -6,13 +6,33 @@ SMF_SET_OPTION(FLATBUFFERS_INSTALL OFF)
 SMF_SET_OPTION(FLATBUFFERS_BUILD_FLATHASH OFF)
 add_subdirectory(flatbuffers)
 
+find_package(fmt 5.2.1 QUIET)
+if(NOT fmt_FOUND)
+  message(STATUS "Could not find fmt, will build it")
+  add_subdirectory(fmt)
+elseif(fmt_VERSION VERSION_GREATER_EQUAL 5.2.1)
+  message(WARNING "Could NOT find fmt: "
+    "Found unsuitable version \"${fmt_VERSION}\", "
+    "but required is at most \"5.2.1\" (found ${fmt_INCLUDE_DIR}). "
+    "Will build it")
+  add_subdirectory(fmt)
+endif()
+
 set(SEASTAR_USER_CFLAGS
   -Wno-stringop-overflow
   -Wno-array-bounds
   -Wno-stringop-truncation
   -Wno-format-overflow
   CACHE STRING "User cflags for seastar deps")
 SMF_SET_OPTION(SEASTAR_ENABLE_TESTS OFF)
+
+# wtf does this do and why does it work
+macro(find_package name)
```

Comment:
haha! did you just dream this up and it worked? may be a ref /link here would be good.

Reply:
heh... this is lifted from ceph. it seems like they had the exact same issue. without that macro, seastar doesn't pick up the version of fmt being built by us.

Reply:
... but i still don't understand it.

Reply:
got it. can you make a comment w/ a ref to the ceph cmake & say that w/out that fmt version is not picked up. that's the info that is critical.

Reply:
yeh. i'm gonna ping kefu and ask about this. google has been completely useless for figuring that out